### PR TITLE
Fixes the author list generation from harvested xml files

### LINKED ISSUE #2800: workflows: process all author XML files
## Expected Behavior

Sometimes there are two XML author list files in an arXiv submission (e.g. https://inspirehep.net/record/1623908/ ). In that case, all authors should be processed by https://github.com/inspirehep/inspire-next/blob/c780ed7ad6f0ac10358f46b771a39cd0eda12d92/inspirehep/modules/workflows/tasks/arxiv.py#L175-L222 instead of only the authors from one of the lists.

## Current Behavior

Only the authors from one of the list are processed.

## Steps to Reproduce (for bugs)

1. run the task on the given arXiv record
2. see that ~half the authors are missing.

### PR DESCRIPTION
**Pull request by david-caro**
Before we were plainly overwriting the record information with a
record generated only by the first authors xml file found.

Fixes #2800

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by david-caro** [comment]
ci test please

**Comment by david-caro** [comment]
Tested on QA, this seems to be working 👍, and fixes the schema issues.

File: inspirehep/modules/workflows/tasks/arxiv.py [hunk 1]
```diff
@@ -235,45 +235,52 @@ def _author_list(obj, eng):
         filename = secure_filename('{0}.tar.gz'.format(arxiv_id))
         tarball = obj.files[filename]
 
-        if tarball:
-            with TemporaryDirectory(prefix='author_list') as scratch_space:
-                tarball_file = retrieve_uri(
+        if not tarball:
+            obj.log.info(
+                'Skipping author list extraction, no tarball with nam "%s" found' % filename
```
Comment (ammirate) [inline inspirehep/modules/workflows/tasks/arxiv.py]: ops typo 'nam'

Comment (david-caro) [reply inspirehep/modules/workflows/tasks/arxiv.py]: 👍

**Pull request merged**